CMakeLists.txt

```diff
@@ -338,6 +338,11 @@ if(NOT ("${ITK_NIFTI_IO_ANALYZE_FLAVOR}" STREQUAL "ITK4Warning" OR
 endif()
 mark_as_advanced(ITK_NIFTI_IO_ANALYZE_FLAVOR)
 
+# Allow non-orthogonal rotation matrix approximation in NIFTI files
+option(ITK_NIFTI_IO_SFORM_PERMISSIVE "Allow non-orthogonal rotation matrix in NIFTI sform" ON)
+mark_as_advanced(ITK_NIFTI_IO_SFORM_PERMISSIVE)
```

Comment:
Should this be a compile time flag or a runtime environment variable?

Reply:
It looks like that he option changes the default behavior, not a final behavior:
`, m_SFORM_Permissive{ ITK_NIFTI_IO_SFORM_PERMISSIVE }`
m_SFORM_Permissive can be changed later `itkSetMacro(SFORM_Permissive, bool);
`

Also I can hardly believe that `#cmakedefine ITK_NIFTI_IO_SFORM_PERMISSIVE` **will work at all**.
```
#cmakedefine VAR ...
will be replaced with either
#define VAR ...
or
/* #undef VAR */
```

So even if the option is set to "ON", it will **not** set _m_SFORM_Permissive_ to _true_ here
`, m_SFORM_Permissive{ ITK_NIFTI_IO_SFORM_PERMISSIVE }`,
it will expand to `, m_SFORM_Permissive{}` and set it to _false_.

If it is not defined, the build will fail.

There is `#cmakedefine01`  s. [configure_file](https://cmake.org/cmake/help/latest/command/configure_file.html), this may work. But, IMHO, it would be better to review the whole thing. The option seems to only set the default, but it is not clear from the description.

Reply:
ok, I changed `#cmakedefine` to `#cmakedefine01` and updated option description.